request.js

```diff
@@ -921,7 +921,7 @@ Request.prototype.onRequestResponse = function (response) {
     })
 
     var responseContent
-    if (self.gzip) {
+    if (self.gzip && self.method !== 'HEAD') {
```

Comment:
Unfortunately there is no reliable way to determine upfront if the response body is empty. And in fact the error is thrown because the gzip content is invalid which may happen even if you have a body.
So at least we can do is forward the error events so that the user can handle them accordingly. Which means that you will receive an error when the request method is GET, the gzip option is true, and there is no response body.